tests/test_file_source.py

```diff
@@ -0,0 +1,141 @@
+
+import cudf
+
+from _utils import TEST_DIRS
+from morpheus.common import FileTypes
+from morpheus.messages.message_meta import MessageMeta
+from morpheus.pipeline.pipeline import Pipeline
+from morpheus.stages.input.file_source import FileSource
+from morpheus.stages.output.in_memory_sink_stage import InMemorySinkStage
+
+
+@pytest.fixture(name="files", scope="function")
+def files_fixture():
+
+    return fsspec.open(os.path.join(TEST_DIRS.tests_data_dir, "filter_probs.json"))
+
+
+@pytest.mark.use_python
+def test_constructor(config):
+    file_source = FileSource(
+        config,
+        files=["path/to/*.json"],
+        watch=True,
+        sort_glob=True,
+        recursive=False,
+        queue_max_size=256,
+        batch_timeout=10.0,
+        file_type=FileTypes.JSON,
+        repeat=3,
+        filter_null=False,
+        parser_kwargs={"key": "value"},
+        watch_interval=2.0,
+    )
+
+    assert file_source._files == ["path/to/*.json"]
+    assert file_source._watch
+    assert file_source._sort_glob
+    assert not file_source._recursive
+    assert file_source._queue_max_size == 256
+    assert file_source._batch_timeout == 10.0
+    assert file_source._file_type == FileTypes.JSON
+    assert not file_source._filter_null
+    assert file_source._parser_kwargs == {"key": "value"}
+    assert file_source._watch_interval == 2.0
+    assert file_source._repeat_count == 3
+
+
+@pytest.mark.use_python
+@pytest.mark.parametrize("input_files", [["file1.json", "file2.json"], []])
+def test_constructor_with_invalid_params(config, input_files):
+    with pytest.raises(ValueError):
+        # 'watch' is True, but multiple files are provided
+        FileSource(config, files=input_files, watch=True)
+
+
+@pytest.mark.parametrize("input_files", [["file1.json", "file2.json"]])
+def test_convert_to_fsspec_files(input_files):
+    actual_output = FileSource.convert_to_fsspec_files(files=input_files)
+
+    assert isinstance(actual_output, fsspec.core.OpenFiles)
+    assert os.path.basename(actual_output[0].full_name) == input_files[0]
+    assert os.path.basename(actual_output[1].full_name) == input_files[1]
+
+
+@pytest.mark.use_python
+@pytest.mark.parametrize(
+    "input_file,filetypes,filter_null,parser_kwargs, repeat_count, expected_count, expected_df_count",
+    [("filter_probs.json", FileTypes.Auto, False, {
+        "lines": False
+    }, 1, 1, 20), ("filter_probs.csv", FileTypes.CSV, False, {}, 2, 2, 20),
+     ("filter_probs.jsonlines", FileTypes.JSON, False, {
+         "lines": True
+     }, 1, 1, 20)])
+def test_generate_frames(input_file,
+                         filetypes,
+                         filter_null,
+                         parser_kwargs,
+                         repeat_count,
+                         expected_count,
+                         expected_df_count):
+    in_file = fsspec.open(os.path.join(TEST_DIRS.tests_data_dir, input_file))
+
+    metas = FileSource.generate_frames(file=in_file,
```

Comment:
Question: Why test `generate_frames` when we could add more tests to `FileSource`?

Reply:
Just an additional test.